eureka2-core/src/main/java/com/netflix/eureka2/utils/rx/RetryStrategyFunc.java

```diff
@@ -0,0 +1,48 @@
+package com.netflix.eureka2.utils.rx;
+
+import rx.Observable;
+import rx.functions.Func1;
+import rx.functions.Func2;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A Func1 with retry options for use in .retryWhen(new RetryStrategyFunc())
+ *
+ * @author David Liu
+ */
+public class RetryStrategyFunc implements Func1<Observable<? extends Throwable>, Observable<Long>> {
```

Comment:
Have you seen ExponentialBackoff or Retrys from ocelli? There are many standard algorithms implemented that we could reuse.